### PR TITLE
add a simple STP decoder

### PR DESCRIPTION
**Pull request by nplanel**

### CODE REVIEW

File: layers/stp.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"github.com/google/gopacket"
+)
+
+// Spanning Tree Protocol is on top of LLC to transport BPDU (bridge protocol data unit) message.
```
Comment (gconnell) [inline layers/stp.go]: Godoc comments should start with the name of the thing they're commenting on.  For here:

```go
// STP decodes spanning tree protocol packets to ...
```

File: layers/llc.go [hunk 2]
```diff
@@ -62,6 +62,9 @@ func decodeLLC(data []byte, p gopacket.PacketBuilder) error {
 	if l.DSAP == 0xAA && l.SSAP == 0xAA {
```
Comment (gconnell) [inline layers/llc.go]: might be better to make this a switch:

```go
switch {
  case l.DSAP == 0xAA && l.SSAP == 0xAA:
    ...
  case l.DSAP == 0x42 && l.SSAP == 0x42:
    ...
}
```

I expect we'll add things here later on, and a switch seems more useful for doing that.

File: layers/stp.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+type STP struct {
+	BaseLayer
```
Comment (gconnell) [inline layers/stp.go]: I'm assuming the idea is to fill this in with actual STP fields at a later date.  Maybe add a TODO here?
```go
// TODO:  parse the STP protocol into actual subfields.
```

File: layers/stp.go (hunk 1 above)
Comment (nplanel) [reply layers/stp.go]: done

File: layers/stp.go (hunk 3 above)
Comment (nplanel) [reply layers/stp.go]: done

File: layers/llc.go (hunk 2 above)
Comment (nplanel) [reply layers/llc.go]: done

**Comment by gconnell** [comment]
Cool, thanks!  Can't wait to see it expanded to decode protocol fields :)

**Pull request merged**